protocols/directfund/directfund.go

```diff
@@ -276,21 +275,13 @@ func (o *Objective) Update(p protocols.ObjectivePayload) (protocols.Objective, e
 	return &updated, nil
 }
 
-// UpdateWithChainEvent updates the objective with observed on-chain data.
-//
-// Only Channel Deposit events are currently handled.
+// UpdateWithChainEvent updates the underlying channel with observed on-chain data.
 func (o *Objective) UpdateWithChainEvent(event chainservice.Event) (protocols.Objective, error) {
 	updated := o.clone()
-
-	de, ok := event.(chainservice.DepositedEvent)
-	if !ok {
-		return &updated, fmt.Errorf("objective %+v cannot handle event %+v", updated, event)
-	}
-	if de.BlockNum > updated.latestBlockNumber {
-		updated.C.OnChainFunding[de.Asset] = de.NowHeld
-		updated.latestBlockNumber = de.BlockNum
+	_, err := updated.C.UpdateWithChainEvent(event)
+	if err != nil {
+		return &Objective{}, err
 	}
-
 	return &updated, nil
 }
 
```

Comment:
There's an argument for removing this entirely. It is useful for unit testing though, which is why I left it in place for now.

Reply:
Is the unit-testing utility more or less that existing tests hit this method, and refactoring them to achieve their goals differently is low-reward work?

Reply:
Yes that's about it. I think it's worth doing that work though, so perhaps not a low reward in the end.

Reply:
[b840a98](https://github.com/statechannels/go-nitro/pull/1540/commits/b840a98ba67c0748434a6e6c1ebffb8953fb4652) and [d174c0f](https://github.com/statechannels/go-nitro/pull/1540/commits/d174c0f7f4a5b01fadcc9cb4709c6a1b98469f86)